backpack/utils/conv_transpose.py

```diff
@@ -33,14 +33,21 @@ def extract_weight_diagonal(module, input, grad_output, N):
 
     input_reshaped = input.reshape(M, -1, spatial_out_numel)
 
+    V_axis, N_axis = 0, 1
     grad_output_viewed = separate_channels_and_pixels(module, grad_output)
     AX = einsum("nkl,vnml->vnkm", (input_reshaped, grad_output_viewed))
-    weight_diagonal = (AX ** 2).sum([0, 1]).transpose(0, 1)
-    weight_diagonal = weight_diagonal.reshape(in_channels, out_channels, *k)
-    return weight_diagonal.transpose(0, 1)
+    N = AX.shape[N_axis]
+    sum_dims = [V_axis, N_axis] if sum_batch else [V_axis]
+    transpose_dims = (V_axis, N_axis) if sum_batch else (V_axis + 1, N_axis + 1)
+    weight_diagonal = (AX ** 2).sum(sum_dims).transpose(*transpose_dims)
+    if sum_batch:
+        weight_diagonal = weight_diagonal.reshape(in_channels, out_channels, *k)
+    else:
+        weight_diagonal = weight_diagonal.reshape(N, in_channels, out_channels, *k)
```

Comment:
`in_channels, out_channels, *k` can be replaced with `module.weight.shape`

Reply:
Hi, we cant do this because we are shifting the channels for `conv_transpose`